target/linux/mediatek/dts/mt7622-iptime-ax8004m-ubi.dts

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR MIT)
+
+/dts-v1/;
+#include "mt7622-iptime-ax8004m.dtsi"
+
+/ {
+	model = "ipTIME AX8004M (UBI)";
+	compatible = "iptime,ax8004m-ubi", "mediatek,mt7622";
+};
+
+&snand {
+	mediatek,bmt-v2;
```

Comment:
You have to limit the range of BMT when using it together with UBI, ie add something like
```
mediatek,bmt-table-size = <0x1000>;
mediatek,bmt-remap-range = <0x0 0x2600000>;
```

Reply:
I already checked the implementation of ```bmt-remap-range```, But If device has many bad blocks (my device actually does), the "kernel" and "ubi" partitions will overlap.

Reply:
You should hence limit the range in such a way, that UBI part is never touched by BMT. Using UBI on top of BMT will not work well, esp. in case of new bad blocks appearing.

Reply:
Yes, but I'm not sure how much to limit. Do you have any good ideas?

Reply:
```
mediatek,bmt-remap-range = <0x0 0x2600000>;
```
Would be it according to my understanding.

Reply:
> ```
> mediatek,bmt-remap-range = <0x0 0x2600000>;
> ```

It actually overlaps the kernel and ubi partitions.

Reply:
It has to overlap with kernel partition, it should not overlap with UBI which according to DTS starts at `0x2600000`.

Reply:
In case: ```mediatek,bmt-remap-range = <0x0 0x2600000>;```
IO request to the logical address 0x2500000 (kernel partition)  can be mapped to physical address 0x2600000 by BMT. 
However, ubi partition is disabled bmt, so logical address 0x2600000 is mapped to physical address 0x2600000 . so it overlaps.

Reply:
So make it `mediatek,bmt-remap-range = <0x0 0x2500000>;` then if that helps to keep BMT off the UBI area.

Reply:
Yes. as you say, adding "margin" may be could be a solution. but I'm not sure how much margin is needed. It will probably depend on the number of bad blocks in the front ubi partition.

Reply:
First of all, shall I add a margin that works on my device appropriately?